modules/RTC/TraceablePeerConnection.js

```diff
@@ -1123,117 +1130,80 @@ TraceablePeerConnection.prototype._removeRemoteTrack = function(toBeRemoved) {
-     * @type {Map<string,TrackSSRCInfo>}
-     */
+TraceablePeerConnection.prototype._processAndExtractSourceInfo = function(localSDP) {
     const ssrcMap = new Map();
 
-    /**
-     * Groups mapped by primary SSRC number
-     * @type {Map<number,Array<SSRCGroupInfo>>}
-     */
-    const groupsMap = new Map();
-
-    if (typeof desc !== 'object' || desc === null
-        || typeof desc.sdp !== 'string') {
-        logger.warn('An empty description was passed as an argument');
-
+    if (!localSDP || typeof localSDP !== 'string') {
         return ssrcMap;
     }
-
-    const session = transform.parse(desc.sdp);
-
-    if (!Array.isArray(session.media)) {
-        return ssrcMap;
-    }
-
-    let media = session.media;
-
-    media = media.filter(mline => mline.direction === MediaDirection.SENDONLY
+    const session = transform.parse(localSDP);
+    const media = session.media.filter(mline => mline.direction === MediaDirection.SENDONLY
         || mline.direction === MediaDirection.SENDRECV);
 
-    let index = 0;
-
-    for (const mLine of media) {
-        if (!Array.isArray(mLine.ssrcs)) {
-            continue; // eslint-disable-line no-continue
-        }
+    if (!Array.isArray(media)) {
+        return;
+    }
 
-        if (Array.isArray(mLine.ssrcGroups)) {
-            for (const group of mLine.ssrcGroups) {
-                if (typeof group.semantics !== 'undefined' && typeof group.ssrcs !== 'undefined') {
-                    // Parse SSRCs and store as numbers
-                    const groupSSRCs = group.ssrcs.split(' ').map(ssrcStr => parseInt(ssrcStr, 10));
-                    const primarySSRC = groupSSRCs[0];
+    for (const localTrack of this.localTracks.values()) {
+        const sourceName = localTrack.getSourceName();
+        const trackIndex = getSourceIndexFromSourceName(sourceName);
+        const mediaType = localTrack.getType();
+        const mLines = media.filter(m => m.type === mediaType);
+        const ssrcGroups = mLines[trackIndex].ssrcGroups;
+        let ssrcs = mLines[trackIndex].ssrcs;
+
+        if (ssrcs?.length) {
+            // Filter the ssrcs with 'cname' attribute.
+            ssrcs = ssrcs.filter(s => s.attribute === 'cname');
+
+            const msid = `${this.rtc.getLocalEndpointId()}-${mediaType}-${trackIndex}`;
+            const ssrcInfo = {
+                ssrcs: [],
+                groups: [],
+                msid
+            };
 
-                    // Note that group.semantics is already present
-                    group.ssrcs = groupSSRCs;
+            ssrcs.forEach(ssrc => ssrcInfo.ssrcs.push(ssrc.id));
 
-                    // eslint-disable-next-line max-depth
-                    if (!groupsMap.has(primarySSRC)) {
-                        groupsMap.set(primarySSRC, []);
-                    }
-                    groupsMap.get(primarySSRC).push(group);
+            if (Array.isArray(ssrcGroups)) {
```

Comment:
We check `ssrcs` with `if (ssrcs?.length) {` but here we do it differently. I feel we should check consistently since they are both arrays.